geomesa-filter/src/test/scala/org/locationtech/geomesa/filter/FilterHelperTest.scala

```diff
@@ -41,6 +44,31 @@ class FilterHelperTest extends Specification {
 
   "FilterHelper" should {
 
+    "optimize the inArray function" >> {
+      val filter = ff.function("inArray", ff.literal("foo"), ff.literal(Array("foo", "bar")))
+      val filter2 = ff.equals(
+        ff.function("inArray", ff.property("name"), ff.literal(new util.ArrayList(util.Arrays.asList("foo", "bar")))),
+        ff.literal(java.lang.Boolean.TRUE)
+      )
+
+      val sf = new ScalaSimpleFeature(sft, "1",
+        Array(new Date(), new Integer(1), new Integer(2), new Integer(3), new Integer(4), WKTUtils.read("POINT(1 5)"), "foo"))
```

Comment:
could we add an `.evaluate(sf)` on some of the test filters to verify that they return correctly?